### PR TITLE
Add Auth0 signIn in test app

### PR DESCRIPTION
**Pull request by jwallra**
Things to consider before you submit the PR:

* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Implement Auth0 signIn/signOut in Sasquatch Puppet, Swift, and Swift-Preview

## Misc

[AB#67959](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/67959)

### CODE REVIEW

File: Sasquatch/SasquatchObjC/AppCenterDelegateObjC.m [hunk 1]
```diff
@@ -71,6 +71,22 @@ - (NSString *)appSecretB2C {
 #endif
 }
 
+- (NSString *)appSecretFirebase {
+#if GCC_PREPROCESSOR_MACRO_PUPPET
+  return kMSPuppetFirebaseAppSecret;
+#else
+  return @"firebase blah";
```
Comment (jwallra) [inline Sasquatch/SasquatchObjC/AppCenterDelegateObjC.m]: The naming for the macro and the other constants is kind of unclear so I'm not sure what these are supposed to be.

**Comment by manuchandel** [comment]
You might want to add these secrets to infolist. I had to do it, since it wont work without it #1779

**Comment by jaelim-ms** [comment]
@manuchandel Thanks for the comment. It is still "work in progress" Auth0 and Firebase implementations are not placed yet. It will be incremental implementations so changes will come later if required.

File: Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift [hunk 2]
```diff
@@ -268,6 +277,18 @@ class MSMainViewController: UITableViewController, AppCenterProtocol {
       UserDefaults.standard.set(text, forKey: kMSAppSecret)
       self.appSecret.text = text
     })
+    let firebaseAction = UIAlertAction(title:"Firebase Secret", style: .default, handler: {
+      (_ action : UIAlertAction) -> Void in
+      let text = self.appCenter.appSecretFirebase();
```
Comment (jwallra) [inline Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift]:
```suggestion
      let text = self.appCenter.appSecretFirebase()
```

File: Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift [hunk 3]
```diff
@@ -268,6 +277,18 @@ class MSMainViewController: UITableViewController, AppCenterProtocol {
...
+      UserDefaults.standard.set(text, forKey: kMSAppSecret)
+      self.appSecret.text = text
+    })
+    let auth0Action = UIAlertAction(title:"Auth0 Secret", style: .default, handler: {
+      (_ action : UIAlertAction) -> Void in
+      let text = self.appCenter.appSecretAuth0();
```
Comment (jwallra) [inline Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift]:
```suggestion
      let text = self.appCenter.appSecretAuth0()
```

File: Sasquatch/Sasquatch/Auth/Auth0Provider.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  private var credentialsManager = CredentialsManager(authentication: Auth0.authentication())
+
+  func signIn(_ completionHandler: @escaping (MSUserInformation?, Error?) -> Void) {
+    Auth0.webAuth().scope("openid offline_access").audience(audienceDomain).start {
+      switch $0 {
+      case .failure(let error):
+        completionHandler(nil, error)
+      case .success(let credentials):
+        let userInformation = MSUserInformation()
+        userInformation.accessToken = credentials.accessToken
```
Comment (guperrot) [inline Sasquatch/Sasquatch/Auth/Auth0Provider.swift]: this is not JWT, what are we doing with it?

Comment (jaelim-ms) [reply Sasquatch/Sasquatch/Auth/Auth0Provider.swift]: Nothing.

Comment (jaelim-ms) [reply Sasquatch/Sasquatch/Auth/Auth0Provider.swift]: It will be removed in another PR.

**Pull request merged**